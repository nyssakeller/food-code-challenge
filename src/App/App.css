@import url('https://fonts.googleapis.com/css?family=Yantramanav:300');

.App {
  font-family: 'Yantramanav', sans-serif;
  font-weight: 300;
  text-align: center;
}

h2 {
  font-size: 28px;
}

