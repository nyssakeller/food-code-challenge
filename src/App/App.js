import React, { Component } from 'react';
import AddFoodForm from '../AddFoodForm/AddFoodForm';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <AddFoodForm />
      </div>
    );
  }
}

export default App;
