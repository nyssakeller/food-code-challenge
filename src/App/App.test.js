import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <App /> );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an initial state of favoriteFoodArray set to an empty array', () => {
    expect(wrapper.state().favoriteFoodArray).toEqual([]);
  });

  it('should update state of favoriteFoodArray when addFoodToFavorites is called', () => {
    wrapper.instance().addFoodToFavorites('chicken nuggets');
    wrapper.instance().addFoodToFavorites('macaroni and cheese');

    expect(wrapper.state().favoriteFoodArray.length).toEqual(2);
  });

});


