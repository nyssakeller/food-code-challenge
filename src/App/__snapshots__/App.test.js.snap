// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          What's Your Favorite Food?
        </h1>,
        <AddFoodForm
          addFoodToFavorites={[Function]}
        />,
        <h2>
          You have no items.
        </h2>,
        <CardContainer
          favoriteFoodArray={Array []}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "What's Your Favorite Food?",
        },
        "ref": null,
        "rendered": "What's Your Favorite Food?",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addFoodToFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no items.",
        },
        "ref": null,
        "rendered": "You have no items.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoriteFoodArray": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            What's Your Favorite Food?
          </h1>,
          <AddFoodForm
            addFoodToFavorites={[Function]}
          />,
          <h2>
            You have no items.
          </h2>,
          <CardContainer
            favoriteFoodArray={Array []}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "What's Your Favorite Food?",
          },
          "ref": null,
          "rendered": "What's Your Favorite Food?",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFoodToFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no items.",
          },
          "ref": null,
          "rendered": "You have no items.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoriteFoodArray": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
